crates/ruff_linter/src/preview.rs

```diff
@@ -0,0 +1,82 @@
+//! Helpers to test if a specific preview style is enabled or not.
+//!
+//! The motivation for these functions isn't to avoid code duplication but to ease promoting preview behavior
+//! to stable. The challenge with directly checking the `preview` attribute of [`LinterSettings`] is that it is unclear
+//! which specific feature this preview check is for. Having named functions simplifies the promotion:
+//! Simply delete the function and let Rust tell you which checks you have to remove.
+
+use crate::settings::LinterSettings;
+
+pub(crate) const fn is_semantic_errors_enabled(settings: &LinterSettings) -> bool {
+    settings.preview.is_enabled()
+}
```

Comment:
Nit: It would be nice if each function had a link to the PR/issue that introduced/documents the new preview behavior. 

Adding it now would mainly be nice to set a good example for future added preview features